internal/pkg/exec/docker.go

```diff
@@ -159,3 +172,51 @@ func imageName(uri, tag string) string {
 	}
 	return fmt.Sprintf("%s:%s", uri, tag)
 }
+
+// IsEcrCredentialHelperEnabled return true if ecr-login is enabled either globally or registry level
+func (c DockerCommand) IsEcrCredentialHelperEnabled(uri string) bool {
+	// Make sure the program is able to obtain the home directory
+	splits := strings.Split(uri, "/")
+	if c.homePath != "" && len(splits) > 0 {
```

Comment:
Hmm why not just 
```suggestion
	homePath := userHomeDirectory()
	if homePath != "" && len(splits) > 0 {
```
I don't see `homePath` being used in the other places.

Reply:
Its done to enable unit testing, so part of unit test code we can control this path, like below:

```go
s := DockerCommand{
  runner: mockRunner,
  buf:    tc.inBuffer,
  homePath: "test/copilot",
}
```